### ISSUE TITLE
Flask Streaming Jsonify

### ISSUE THREAD

**Issue by BrendanBall** [mitsuhiko/flask#1569]
Jsonify doesn't work with streaming in flask.
If I have a generator in a method that calls `yield jsonify({'test':'test'})`

and I call the generator with 
`return Response(stream_with_context(generator()), mimetype='application/json')`

then flask throws an error saying `application must write bytes`. 
For now I have resorted to using plain python `json.dumps(dict)`

**Comment by mitchfriedman** [comment]
I may be mistaken, but I think jsonify returns a Response object, so trying to create another Response with a Response as the first argument will fail, as expected.  On the other hand, json.dumps(dict) returns a string, as expected, so creating a Response with a string works correctly

**Comment by aftalavera** [comment]
It will be a matter of pulling the data aout of the response object returned by jsonify like:

yield jsonify({'test':'test'}).data

**Comment by nicorevin** [comment]
This is not Flask's issue - https://bugs.python.org/issue14573

**Comment by davidism** [comment]
`jsonify` is for producing a response with JSON data, use `flask.json.dumps` to just produce the JSON data.  Your use case doesn't make sense though, as you're not streaming anything: the JSON is generated and returned in a single yield.  It would be equivalent to just return `jsonify` directly.

**Comment by SumNeuron** [comment]
I would like to reopen this issue, but under another premise. I am using event stream and yield with flask. All of it is great. I have parse my dictionary into single line json and it works fine. Then, I use python's json.dumps to convert my dictionary into json and the event stream breaks. I know that this sounds like a python issue. However, I have tested hand made json vs python's json.dumps on simple objects e.g. {"foo":"bar"} and this still fails only for python dumps. I have checked for weird quoting and can not spot it. So I would appreciate it if you might look into it. Hand made '{"foo":"bar"}' works, json.dumps({"foo":"bar"}) which gives the same result does not...
